docs/aws.md

```diff
@@ -0,0 +1,150 @@
+# AWS Setup Instructions
+The goal of this document is to provide a guide on how to set up the GitHub Action runner on AWS. This document will guide you through the account setup required to use this action with AWS.
+
+## Prerequisites
+- An AWS account
+
+## Setup
+1. Prepare an IAM user with AWS access keys
+    1. Sign in to your AWS Management Console.
+    2. Go to the IAM console.
+    3. In the navigation pane, choose "Policies" and click "Create Policy".
+    4. Select the "JSON" tab, paste the following JSON, and click "Next":
+        ```json
+        {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Action": [
+                "ec2:RunInstances",
+                "ec2:TerminateInstances",
+                "ec2:DescribeInstances",
+                "ec2:DescribeInstanceStatus"
+              ],
+              "Resource": "*"
+            }
+          ]
+        }
+        ```
+    5. Name the policy (e.g., `gha-runner-policy`) and click "Create Policy".
+    6. In the navigation pane, choose "Users" and click "Add user".
+    7. Enter a username (e.g., `gha-runner-user`) and ensure "Provide user access to the AWS Management Console" is unchecked.
+    8. Select "Attach policies directly", search for the policy you created (`gha-runner-policy`), and select it.
+    9. Click "Create user".
+    10. Click on the newly created user, go to the "Security credentials" tab, and click "Create access key".
+    11. Choose "Other", click "Next", and then copy the Access Key and Secret Access Key (or download the CSV file). These keys will not be shown again.
+2. Create your GitHub Access Token
+    1. This can be done with either a Personal Access Token or a Fine-Grained Personal Access Token.
+    2. Go to your GitHub account settings.
+    3. Click on "Developer settings".
+    4. Create a new token with `repo` scope.
+    5. Save and/or copy the token.
+3. Add your credentials to your repository secrets
+    1. Go to your repository on GitHub.
+    2. Click on "Settings", then "Secrets and Variables", and then "Actions".
+    3. Click "New repository secret".
+    4. Add the following secrets:
+      - `AWS_ACCESS_KEY_ID` - The Access Key you copied earlier.
+      - `AWS_SECRET_ACCESS_KEY` - The Secret Access Key you copied earlier.
+      - `GITHUB_TOKEN` - The GitHub token you copied earlier.
```

Comment:
```suggestion
      - `GH_PAT` - The GitHub token you copied earlier.
```

Reply:
+1 otherwise we're shadowing the `$GITHUB_TOKEN` env var that GitHub sets by default.

Reply:
I came upon this suggestion via a slightly less sophisticated means - I tried to set it verbatim and the page raised an error saying that the token name can't begin with `GITHUB_`